certora/specs/RatioMath.spec

```diff
@@ -182,3 +187,25 @@ rule repayDecreasesBorrowRatio(env e, MorphoHarness.MarketParams marketParams, u
     // Check that the ratio decreases: assetsBefore/sharesBefore >= assetsAfter / sharesAfter
     assert assetsBefore * sharesAfter >= assetsAfter * sharesBefore;
 }
+
+rule liquidateDecreasesBorrowRatio(env e, MorphoHarness.MarketParams marketParams, address borrower, uint256 seizedAssets, uint256 repaidShares, bytes data)
```

Comment:
It seems that you are not requiring anything more than in `onlyAccrueInterestCanIncreaseBorrowRatio`, why not merging then?

Reply:
or it is the `require assetsAfter != 1;` ? why does it break without this ? you know ?

Reply:
Yes it is that require that is needed for the rule to pass, and it corresponds to the case where there is no borrow at all after liquidate (only 1 virtual asset). In case where liquidate removes all borrow, then the exchange rate is reset, which can make so that it increases. But it's fine at this point because there are no users, so it should not impact anybody. This is the same reason why we are checking that the repay all resets the borrow exchange rate [here](https://github.com/morpho-org/morpho-blue/blob/63ead62cd829f4ffb24de5e475f731963451a5a3/certora/specs/ExactMath.spec#L37), instead of checking that it decreases it

Reply:
get it
why withdrawing all shares should not do the same thing?

Reply:
> why withdrawing all shares should not do the same thing?

The same reason that there is an edge case for repay with a `zeroFloorSub` https://github.com/morpho-org/morpho-blue/blob/7298a03c8b7c47d19a8a5f9220b0f12c49735d2a/src/Morpho.sol#L288 that does not exist for withdraw.
In real scenarios, you cannot resest the supply exchange rate, because you cannot remove all assets. This is coming from the fact that the assets to withdraw are rounded down, whereas the assets to repay are rounded up. So when withdrawing all shares, you always increase the exchange rate